client/client.py

```diff
@@ -0,0 +1,100 @@
+"""
+Base class which contains the basic methods to interaction with the write api of NodeBB.
+"""
+import json
+
+import requests
+import urlparse
+from django.conf import settings as django_settings
+
+NODEBB_ADMIN_UID = 1
+
+
+class Client(object):
+    """
+    Client Class responsible to make connection with NodeBB and perform all calls.
+    """
+    def __init__(self):
+        self._configure()
+
+    def _configure(self):
+        self.endpoint = django_settings.NODEBB_SETTINGS['URL']
+        self.admin_uid = NODEBB_ADMIN_UID
+        self.headers = {
+            'Authorization': 'Bearer {}'.format(django_settings.OPENEDX_NODEBB_DISCUSSION['NODEBB_API_TOKEN']),
+            'Content-Type': 'application/json'
+        }
+
+    def _request(self, method, path, **kwargs):
+        """
+        Hidden Method of Client this function will generate all of the Api Call's and return response.
+
+        Args:
+            method (str): Api call method can be Post, Put, and Delete
+            path (str): Api path to make call
+            kwargs (dictionary): All other necessary data to make POST, PUT or DELETE calls to API
+
+        Returns:
+            tuple: Tuple in the form (response_code, json_response) received from requests call.
+        """
+        if '_uid' not in kwargs:
+            kwargs.update({'_uid': self.admin_uid})
+
+        response = requests.request(
+            method,
+            urlparse.urljoin(self.endpoint, path),
+            headers=self.headers,
+            data=json.dumps(kwargs)
+        )
+        status_code, reason = response.status_code, response.reason
+
+        if reason != 'OK':
+            return status_code, reason
+
+        # ValueError occurs when `.json()` reads invalid JSON.
```

Comment:
remove this comment. I don't think it's necessary here.

Reply:
Done